lib/tasks/adhoc.rake

```diff
@@ -0,0 +1,15 @@
+namespace :adhoc do
+  desc 'Extract a list of All articles and what categories they sit in'
+  task :extract_page => :environment do
+    File.open('pages_with_cat.csv','w') do |file|
+      file.write "Url\tCategories\n"
+      en = Comfy::Cms::Site.where(label: :en).first
+      Comfy::Cms::Page.where(site: en, state: :published).find_each do |page|
```

Comment:
But.... I wasn't commenting on the use of `find_each`? My comment was about how you derived the scope before you called `find_each` on it.

So in full it would be: `en.pages.where(state:published).find_each do |page|`

And now I come to think if it, you need to use the `published` scope here, not merely selecting based on the state column.

So: `en.pages.published.find_each do |page|`

There's SOOOO much more to how you find published articles! (https://github.com/moneyadviceservice/cms/pull/324)  #stilltraumatised #neverforget